fix for adjoints with preequilibration and sensitivities

paulstapor opened the pull request:
fix #333

paulstapor commented:
That should already do the job... so let's not wait until Friday, if this is enough

dweindl commented on src/steadystateproblem.cpp:
```diff
@@ -70,8 +70,7 @@ void SteadystateProblem::workSteadyStateProblem(ReturnData *rdata,
     /* Compute steady state sensitvities */
-    if (rdata->sensi_meth == AMICI_SENSI_FSA &&
-        rdata->sensi >= AMICI_SENSI_ORDER_FIRST)
+    if (rdata->sensi >= AMICI_SENSI_ORDER_FIRST)
```
&& rdata->sensimeth != AMICI_SENSI_NONE

paulstapor commented:
Just for the moment, I added the other solver which can also be used for preequilibration and changed the naming in the way we discussed...
I hope things are ready for merging now...
